File: Assets/MediaPipe/SDK/Scripts/External/Glog.cs
namespace Mediapipe {
  public class Glog {
    public enum Severity : int {
      INFO = 0,
      WARNING = 1,
      ERROR = 2,
      FATAL = 3,
    }

    public static void Initialize(string name, string dir) {
      UnsafeNativeMethods.google_InitGoogleLogging__PKc(name, dir).Assert();
    }

    public static void Shutdown() {
      UnsafeNativeMethods.google_ShutdownGoogleLogging().Assert();
    }

    public static void Log(Severity severity, string str) {
      switch (severity) {
        case Severity.INFO: {
          UnsafeNativeMethods.glog_LOG_INFO__PKc(str);
          break;
        }
        case Severity.WARNING: {
          UnsafeNativeMethods.glog_LOG_WARNING__PKc(str);
          break;
        }
        case Severity.ERROR: {
          UnsafeNativeMethods.glog_LOG_ERROR__PKc(str);
          break;
        }
        case Severity.FATAL: {
          UnsafeNativeMethods.glog_LOG_FATAL__PKc(str);
          break;
        }
      }
    }

    public static void FlushLogFiles(Severity severity = Severity.INFO) {
      UnsafeNativeMethods.google_FlushLogFiles(severity);
    }
  }
}

File: C/mediapipe_api/external/glog.cc
#include "mediapipe_api/external/glog.h"

void glog_LOG_INFO__PKc(const char* str) {
  LOG(INFO) << str;
}

void glog_LOG_WARNING__PKc(const char* str) {
  LOG(WARNING) << str;
}

void glog_LOG_ERROR__PKc(const char* str) {
  LOG(ERROR) << str;
}

void glog_LOG_FATAL__PKc(const char* str) {
  LOG(FATAL) << str;
}

void google_FlushLogFiles(google::LogSeverity severity) {
  google::FlushLogFiles(severity);
}
